### PR TITLE
feat(oid4vp): extend extractClientIdPrefix to return clientId and prefix

### PR DESCRIPTION
**Pull request by manuraf**
#### List of Changes

- Introduced a new `ClientIdParts` interface (exported from `packages/oid4vp`) that bundles the resolved `prefix` and the clean `clientId` string:
  ```ts
  export interface ClientIdParts {
    prefix: ClientIdPrefix | string;
    clientId: string;
  }
  ```
- Rewrote `extractClientIdPrefix` to return `ClientIdParts` instead of `ClientIdPrefix`. Parsing now uses `indexOf(":")` so the full substring after the first colon is preserved as `clientId` (handles values like `x509_hash:a:b:c` correctly).
- Unknown prefixes (a colon is present but the left-hand side doesn't match any `ClientIdPrefix` enum value) are now returned as a raw string instead of being silently coerced to `ClientIdPrefix.NONE`.
- Updated both internal call sites to destructure `{ prefix: clientIdPrefix }` — no behavioural change to the existing logic in either site.
- Added a dedicated `describe("extractClientIdPrefix")` block in `parse-authorization-request.test.ts` covering all four mapping rules plus the multi-colon edge case.

#### Motivation and Context

`extractClientIdPrefix` previously returned only the `ClientIdPrefix` enum value, forcing any consumer that also needed the unprefixed `client_id` (e.g. to forward it downstream or for logging) to re-implement the same string-splitting logic. Additionally, any `client_id` that contained a `:` but didn't match a known scheme was silently mapped to `ClientIdPrefix.NONE`, which is semantically incorrect — the prefix was simply discarded.

This change makes the parsed parts directly available from a single call and preserves unknown prefixes faithfully, giving consumers full information without duplication.

#### How Has This Been Tested?

- **Unit tests** – new `describe("extractClientIdPrefix")` block with 5 cases:
  - `x509_hash:<value>` → `{ prefix: ClientIdPrefix.X509_HASH, clientId: "<value>" }`
  - `openid_federation:<value>` → `{ prefix: ClientIdPrefix.OPENID_FEDERATION, clientId: "<value>" }`
  - `unknown_prefix:<value>` → `{ prefix: "unknown_prefix", clientId: "<value>" }`
  - `no_prefix` (no colon) → `{ prefix: ClientIdPrefix.NONE, clientId: "no_prefix" }`
  - `x509_hash:a:b:c` (multi-colon) → `{ prefix: ClientIdPrefix.X509_HASH, clientId: "a:b:c" }`
- **Full test suite** – `pnpm test` passes with 759/759 tests green across all packages.
- **Type check** – `pnpm types:check` passes with no errors.

### CODE REVIEW

File: packages/oid4vp/src/authorization-request/parse-authorization-request.ts [hunk 1]
```diff
@@ -31,19 +31,34 @@ export enum ClientIdPrefix {
...
+  }
+  if (rawPrefix === ClientIdPrefix.OPENID_FEDERATION) {
+    return { clientId: rest, prefix: ClientIdPrefix.OPENID_FEDERATION };
   }
-  return ClientIdPrefix.NONE;
+
+  return { clientId: rest, prefix: rawPrefix };
 }
 
 /**
```
Comment (flaviodelgrosso) [inline packages/oid4vp/src/authorization-request/parse-authorization-request.ts:69]: Could we keep the IT-Wallet profile validation explicit here? The authorization request `client_id` values must use either the `openid_federation` or `x509_hash` prefix. Returning arbitrary prefixes as successful `ClientIdParts` makes the exported helper look like it supports non-profile schemes, and standalone callers such as response creation can proceed unless they add their own validation. I’d consider either rejecting unsupported prefixes in this profile-specific helper, or splitting raw parsing from a stricter IT-Wallet validator.

Comment (manuraf) [reply packages/oid4vp/src/authorization-request/parse-authorization-request.ts:69]: [d5cab91](https://github.com/pagopa/io-wallet-sdk/pull/152/commits/d5cab917bc8148d0e94fcca8cc694cbc8a473e1b)

**Review by flaviodelgrosso — APPROVED** [review]
